test/parallel/test-readline-async-iterators.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const common = require('../common');
+const fs = require('fs');
+const join = require('path').join;
+const readline = require('readline');
+const assert = require('assert');
+
+common.crashOnUnhandledRejection();
+
+const tmpdir = require('../common/tmpdir');
+tmpdir.refresh();
+
+const filename = join(tmpdir.path, 'test.txt');
+const file_content = 'abc123\n' +
+                     '南越国是前203年至前111年存在于岭南地区的一个国家\n';
+
+fs.writeFile(filename, file_content);
+
+async function tests() {
+  await (async function() {
+    const readable = fs.createReadStream(filename);
+    const rli = readline.createInterface({
+      input: readable,
+      crlfDelay: Infinity
+    });
+
+    let data = '';
+    for await (const k of rli) {
+      data += k;
+    }
+
+    assert.strictEqual(data, file_content);
+  })();
+}
+
+// to avoid missing some tests if a promise does not resolve
+tests().then(common.mustCall());
+
+process.on('exit', function() {
+  fs.unlinkSync(filename);
```

Comment:
Can't this instead be moved to the `common.mustCall()` callback?

Reply:
It can be removed entirely. Test files do not need to clean up the temp directory. Tests that use the temp directory clean it up themselves at the start of the test with `tmpdir.refresh()`.

Reply:
Should I also clean up fs tests in separate pr? [example](https://github.com/nodejs/node/blob/master/test/parallel/test-fs-write-file.js#L106)